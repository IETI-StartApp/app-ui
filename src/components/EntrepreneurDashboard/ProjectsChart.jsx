import * as React from 'react';
import Paper from '@material-ui/core/Paper';
import {
  Chart,
  BarSeries,
  ArgumentAxis,
  ValueAxis,
} from '@devexpress/dx-react-chart-material-ui';
import { Animation } from '@devexpress/dx-react-chart';


export const ProjectsChart = () => {
    const data= [
      {name: "Proyecto 1", financed: 75}, 
      {name: "Proyecto 2", financed: 57}, 
      {name: "Proyecto 3", financed: 67},
      {name: "Proyecto 4", financed: 17},
      {name: "Proyecto 5", financed: 2},
    ];

    return (
      <Paper>
        <Chart
          data={data}
        >
          <ArgumentAxis />
          <ValueAxis />
          <BarSeries
            barWidth={0.3}
            valueField="financed"
            argumentField="name"
            color="#950740"
          />
          <Animation />
        </Chart>
      </Paper>
    );
}