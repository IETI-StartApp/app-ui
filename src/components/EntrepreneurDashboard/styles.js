import { makeStyles } from '@material-ui/core';

export const useStyles = makeStyles((theme) => ({
      fab: {
        position: 'fixed',
        bottom: theme.spacing(5),
        right: theme.spacing(5),
        backgroundColor: '#950740',
        color: '#fff'
    },
}));