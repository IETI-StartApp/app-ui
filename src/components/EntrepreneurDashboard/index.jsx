import React from 'react';
import AttachMoneyIcon from '@material-ui/icons/AttachMoney';
import StarBorderIcon from '@material-ui/icons/StarBorder';
import { Dashboard } from '../Dashboard/index';
import { ProjectsChart } from './ProjectsChart';
import PermIdentityIcon from '@material-ui/icons/PermIdentity';
import CheckIcon from '@material-ui/icons/Check';

export const EntrepreneurDashboard = () => {
    const items = [
        { number: "5", description: "Proyectos Creados", icon: <StarBorderIcon fontSize="large" />, path: "/" },
        { number: "4.5M", description: "Capital recaudado", icon: <AttachMoneyIcon fontSize="large" />, path: "/dashboard" },
        { number: "17", description: "Socios", icon: <PermIdentityIcon fontSize="large" />, path: "/" },
        { number: "0", description: "Proyectos exitosos", icon: <CheckIcon fontSize="large" />, path: "/" }
    ];
    const projects = [
        {
            name: "Proyecto 1", 
            image:"https://www.binarias.org/wp-content/uploads/2020/12/tegnologia.jpeg", 
            finance: { value:15000000, valuation:20000000, startDate: "10/03/2021"}
        },
        {
            name: "Proyecto 2", 
            image:"https://s3-us-west-2.amazonaws.com/wp-mpro-blog/wp-content/uploads/2016/03/14133959/s3-blog-tecnologia-como-prioridad-min.png", 
            finance: { value:10000000, valuation:17500000, startDate: "12/02/2021"},
        },
        {
            name: "Proyecto 3", 
            image:"https://thelogisticsworld.com/wp-content/uploads/2020/02/empresario-analizando-informe-financiero-empresa-graficos-realidad-aumentada_34141-360.jpg", 
            finance: { value:8000000, valuation:12000000, startDate: "21/01/2021"},
        },
    ];

    return (
        <Dashboard items={ items } projects={ projects } chart={ <ProjectsChart /> } projectsTitle={ "Mis proyectos" } chartTitle={ "Mis proyectos: porcentaje financiado" } />
    );
}