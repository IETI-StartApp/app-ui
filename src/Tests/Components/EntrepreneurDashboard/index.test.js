import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import React from 'react';
import { EntrepreneurDashboard } from '../../../components/EntrepreneurDashboard/index';
import { Fab } from '@material-ui/core/';

describe('Testing Entrepreneur Dashboard', () => {

    const wrapper = shallow(<EntrepreneurDashboard />)

    test('should load Entrepreneur Dashboard', () => {
        expect(wrapper).toMatchSnapshot();
    });


    test('should call handleDrawerToggle function', () => {
        const iconButton = wrapper.find(Fab).at(0);
        iconButton.simulate("Click");
    });
})
