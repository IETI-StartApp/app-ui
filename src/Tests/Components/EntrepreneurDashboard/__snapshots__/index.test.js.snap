// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Entrepreneur Dashboard should load Entrepreneur Dashboard 1`] = `
<Fragment>
  <Dashboard
    chart={<ProjectsChart />}
    chartTitle="Mis proyectos: porcentaje financiado"
    items={
      Array [
        Object {
          "description": "Proyectos Creados",
          "icon": <Memo
            fontSize="large"
          />,
          "number": "5",
          "path": "/",
        },
        Object {
          "description": "Capital recaudado",
          "icon": <Memo
            fontSize="large"
          />,
          "number": "4.5M",
          "path": "/dashboard",
        },
        Object {
          "description": "Socios",
          "icon": <Memo
            fontSize="large"
          />,
          "number": "17",
          "path": "/",
        },
        Object {
          "description": "Proyectos exitosos",
          "icon": <Memo
            fontSize="large"
          />,
          "number": "0",
          "path": "/",
        },
      ]
    }
    projects={
      Array [
        Object {
          "finance": Object {
            "startDate": "10/03/2021",
            "valuation": 20000000,
            "value": 15000000,
          },
          "image": "https://www.binarias.org/wp-content/uploads/2020/12/tegnologia.jpeg",
          "name": "Proyecto 1",
        },
        Object {
          "finance": Object {
            "startDate": "12/02/2021",
            "valuation": 17500000,
            "value": 10000000,
          },
          "image": "https://s3-us-west-2.amazonaws.com/wp-mpro-blog/wp-content/uploads/2016/03/14133959/s3-blog-tecnologia-como-prioridad-min.png",
          "name": "Proyecto 2",
        },
        Object {
          "finance": Object {
            "startDate": "21/01/2021",
            "valuation": 12000000,
            "value": 8000000,
          },
          "image": "https://thelogisticsworld.com/wp-content/uploads/2020/02/empresario-analizando-informe-financiero-empresa-graficos-realidad-aumentada_34141-360.jpg",
          "name": "Proyecto 3",
        },
      ]
    }
    projectsTitle="Mis proyectos"
  />
  <WithStyles(ForwardRef(Fab))
    aria-label="Chat"
    className="makeStyles-fab-1"
    onClick={[Function]}
  >
    <Memo([object Object]) />
  </WithStyles(ForwardRef(Fab))>
  <ChatBotContainer
    onClose={[Function]}
    open={true}
  />
</Fragment>
`;
